### PR TITLE
Migrate to copier

### PR DESCRIPTION
**Pull request by dperl-dls**
To test:
- Check docs are intact
- Confirm changed tests are correct

### CODE REVIEW

File: src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_StartUp_py3v1.py [hunk 1]
```diff
@@ -41,10 +40,11 @@ def read_parameter_file(
 
 @log.log_on_entry
 def fiducials(chip_type: int):
+    fiducial_list: list | None = None
     if chip_type in [ChipType.Oxford, ChipType.OxfordInner, ChipType.Minichip]:
-        fiducial_list: list = []
-        # No fiducial for custom
+        fiducial_list = []
     elif chip_type == ChipType.Custom:
+        # No fiducial for custom
```
Comment (DominicOram) [inline src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_StartUp_py3v1.py:47]: Should: What does this comment give you over the existing log?

File: tests/i04/test_thawing.py [hunk 2]
```diff
@@ -124,9 +122,13 @@ def do_thaw_func():
     ],
 )
 def test_given_different_rotations_and_times_then_velocity_correct(
-    smargon: Smargon, thawer: Thawer, time, rotation, expected_speed
+    smargon: Smargon,
+    thawer: Thawer,
+    time: float | Literal[10] | Literal[50],
+    rotation: Literal[360] | Literal[100] | Literal[-100],
+    expected_speed: ApproxBase | Literal[72] | Literal[4],
```
Comment (DominicOram) [inline tests/i04/test_thawing.py:129]: Should: This odd, if I want to add another case to the parameterize I need to change these types? I feel like they should just be floats/ints?

File: tests/i04/test_thawing.py [hunk 3]
```diff
@@ -141,10 +143,14 @@ def test_given_different_rotations_and_times_then_velocity_correct(
     ],
 )
 def test_given_different_rotations_then_motor_moved_relative(
-    smargon: Smargon, thawer: Thawer, start_pos, rotation, expected_end
+    smargon: Smargon,
+    thawer: Thawer,
+    start_pos: Literal[0] | Literal[78],
+    rotation: Literal[360] | Literal[100] | Literal[-100],
+    expected_end: Literal[360] | Literal[178] | Literal[-100],
```
Comment (DominicOram) [inline tests/i04/test_thawing.py:150]: Should: Same as above

File: tests/i24/serial/setup_beamline/test_zebra_plans.py [hunk 4]
```diff
@@ -29,7 +29,7 @@
 
 
 async def test_arm_and_disarm_zebra(zebra: Zebra, RE):
-    zebra.pc.arm.TIMEOUT = 1
+    zebra.pc.arm.TIMEOUT = 1  # type: ignore
```
Comment (DominicOram) [inline tests/i24/serial/setup_beamline/test_zebra_plans.py:32]: Could: I feel like the correct answer is to change the definition to be `TIMEOUT: float = 3`

**Review by DominicOram — CHANGES_REQUESTED** [review]
Great, thank you! Couple of comments, some of which are more related to general code issues that to this change so feel free to ignore/spin out

File: .copier-answers.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+component_owner: group:data-acquisition
+description: Bluesky tools for MX Beamlines at DLS
+distribution_name: mx_bluesky
+docker: true
+docs_type: sphinx
+git_platform: github.com
+github_org: DiamondLightSource
+package_name: mx_bluesky
+pypi: true
+repo_name: mx_bluesky
```
[the remark this answers is not in the document]
Comment (callumforrester) [reply .copier-answers.yml:15]: It's in [PEP8](https://peps.python.org/pep-0008/#package-and-module-names)

File: src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py [hunk 6]
```diff
@@ -251,22 +248,24 @@ def save_screen_map() -> MsgGenerator:
 
 @log.log_on_entry
 def upload_parameters(
-    chipid: str = "oxford",
-    pmac: PMAC = inject("pmac"),
+    chipid: str = "oxford", pmac: PMAC = inject("pmac"), width: int | None = None
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py:251]: For this and other similar, I didn't want to change things beyond the minimum necessary to make pyright happy here, but I agree these things all need cleaning up

File: src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_Mapping_py3v1.py [hunk 7]
```diff
@@ -79,36 +79,36 @@ def plot_file(fid, chip_type):
     ax1.set_xlim(-1, 26)
     ax1.set_ylim(-1, 26)
     ax1.invert_yaxis()
-    check_files(["%s.png" % chip_type])
-    plt.savefig("%s.png" % fid[:-5], dpi=200, bbox_inches="tight", pad_inches=0.05)
+    check_files("i24", [f"{chip_type}.png"])
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_Mapping_py3v1.py:82]: https://github.com/DiamondLightSource/mx_bluesky/issues/161

File: src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_Manager_py3v1.py:251]: https://github.com/DiamondLightSource/mx_bluesky/issues/160

File: src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_StartUp_py3v1.py (hunk 1 above)
Comment (dperl-dls) [reply src/mx_bluesky/i24/serial/fixed_target/i24ssx_Chip_StartUp_py3v1.py:47]: I didn't add this comment, just moved it down two lines so it was in the section it referred to

File: tests/i04/test_thawing.py (hunk 2 above)
Comment (dperl-dls) [reply tests/i04/test_thawing.py:129]: hah, I guess ruff added these automagically

File: docs/developer/serial-crystallography-on-i24/how-to/run-a-collection.rst [hunk 8]
```diff
@@ -12,7 +12,7 @@ A couple of entry points have been set up so that:
 
 Before opening the experiment specific edm, each of these entry points
 will start a ``BlueAPI`` server. The configuration used by ``BlueAPI``
-is saved in ``src/mx_bluesky/i24/serial/blueapi_config.yaml``.
+is saved in ``src/mx-bluesky/i24/serial/blueapi_config.yaml``.
```
Comment (DominicOram) [inline docs/developer/serial-crystallography-on-i24/how-to/run-a-collection.rst:15]: Should: This probably hasn't changed and I don't think it should

File: docs/developer/serial-crystallography-on-i24/how-to/run-a-collection.rst [hunk 9]
```diff
@@ -75,7 +75,7 @@ set up the coordinate system.
 
 Before this step remember to reset the scale and skew factors as well as
 the motor directions as needed. Current values are saved in
-``src/mx_bluesky/i24/serial/parameters/fixed_target`` in the
+``src/mx-bluesky/i24/serial/parameters/fixed_target`` in the
```
Comment (DominicOram) [inline docs/developer/serial-crystallography-on-i24/how-to/run-a-collection.rst:78]: Should: As above

File: docs/developer/serial-crystallography-on-i24/how-to/stage-pmac-moves.rst [hunk 10]
```diff
@@ -81,8 +81,8 @@ Theory for this computation
 
 The plan needs information stored in a few files:
 
-* The motor directions are stored in ``src/mx_bluesky/i24/serial/parameters/fixed_target/cs/motor_directions.txt.`` The motor number multiplied by the motor direction should give the positive chip direction. 
-* The scale values for x,y,z, the skew value and the sign of Sx, Sy, Sz are all stored in ``src/mx_bluesky/i24/serial/parameters/fixed_target/cs/cs_maker.json``
+* The motor directions are stored in ``src/mx-bluesky/i24/serial/parameters/fixed_target/cs/motor_directions.txt.`` The motor number multiplied by the motor direction should give the positive chip direction. 
+* The scale values for x,y,z, the skew value and the sign of Sx, Sy, Sz are all stored in ``src/mx-bluesky/i24/serial/parameters/fixed_target/cs/cs_maker.json``
```
Comment (DominicOram) [inline docs/developer/serial-crystallography-on-i24/how-to/stage-pmac-moves.rst:85]: Should: I don't think these have changed (and shouldn't change)

File: docs/user/reference/api.rst [hunk 11]
```diff
@@ -1,14 +1,14 @@
...
 
-    ``mx_bluesky``
+    ``mx-bluesky``
     -----------------------------------
 
-This is the internal API reference for mx_bluesky
+This is the internal API reference for mx-bluesky
 
-.. data:: mx_bluesky.__version__
+.. data:: mx-bluesky.__version__
```
Comment (DominicOram) [inline docs/user/reference/api.rst:11]: Should: Not changed

**Review by DominicOram — CHANGES_REQUESTED** [review]
Thanks, I think you were slightly too aggressive on the find/replace. Otherwise good.

**Review by DominicOram — APPROVED** [review]
Great, thank you!

**Pull request merged**